src/main/java/nl/tudelft/cse1110/andy/execution/security/AndySecurityManager.java

```diff
@@ -122,22 +127,31 @@ private boolean checkLoggingPermission(Permission perm) {
         return false;
     }
 
-    private boolean checkRuntimePermission(Permission perm, boolean mockitoInternal) {
+    private boolean checkRuntimePermission(Permission perm, boolean mockitoInternal, boolean seleniumInternal) {
         // Allow various runtime permissions as they are needed in order to execute the tests
         if (perm instanceof RuntimePermission) {
             if (mockitoInternal && checkMockitoInternalRuntimePermission(perm)
-                || checkReflectionRuntimePermissions(perm)
-                || perm.getName().equals("loggerFinder")
-                || perm.getName().equals("createClassLoader")
-                || perm.getName().equals("createSecurityManager")
-                || perm.getName().equals("accessSystemModules")
-                || checkNetworkRuntimePermissions(perm)) {
+                || seleniumInternal && checkSeleniumRuntimePermissions(perm)
+                || checkNormalRuntimePermissions(perm)) {
                 return true;
             }
         }
         return false;
     }
 
+    private boolean checkSeleniumRuntimePermissions(Permission perm) {
+        return perm.getName().equals("modifyThread");
```

Comment:
A comment explaining what "modifythread" is may help future understanding

Reply:
I'm not sure what exactly this is needed for since there's no documentation on the permissions Selenium requires. I added a comment simply explaining that it's necessary for Selenium, is that sufficient?

Reply:
Maybe what modifyThread means. Is this a method in selenium or a Java thing ?

Reply:
It's a Java thing, it's a qualifier of `RuntimePermission`. It allows the process to execute methods such as `interrupt()`, `suspend()`, `resume()` on threads. However, I am not sure why exactly Selenium requires that as it is not specified anywhere. https://docs.oracle.com/en/java/javase/15/security/permissions-jdk1.html

Reply:
Maybe that's the comment then :)